src/xcode/ENA/ENA/Source/AppDelegate & Globals/NotificationManager.swift

```diff
@@ -10,10 +10,20 @@ final class NotificationManager: NSObject, UNUserNotificationCenterDelegate {
 	weak var appDelegate: AppDelegate?
 
 	func userNotificationCenter(_: UNUserNotificationCenter, willPresent notification: UNNotification, withCompletionHandler completionHandler: @escaping (UNNotificationPresentationOptions) -> Void) {
-		// Checkout a event checkin.
-		if notification.request.identifier.contains(EventCheckoutService.notificationIdentifierPrefix) {
+		// Show badge on check-in tab when checkout.
+		if notification.request.identifier.contains(LocalNotificationIdentifier.checkout.rawValue) {
 			appDelegate?.eventCheckoutService.checkoutOverdueCheckins()
 		}
+		
+		// Show badge on certificates tab when certificate is expired.
+		if notification.request.identifier.contains(LocalNotificationIdentifier.certificateExpired.rawValue) {
+			appDelegate?.healthCertificateService.unseenTestCertificateCount.value += 1
```

Comment:
why not overwrite init and inject healthCertificateService?
Also you can restructure the class and add some MARKs as well.

Reply:
- Injected all services and removed the reference to the appDelegate.
- Removed obsolete function
- Added MARKs